### PR TITLE
Added Hyper-parameter tuning: Helmholtz equation over a 2D square domain

### PR DESCRIPTION
**Pull request by pescap**

### CODE REVIEW

File: examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+"""Backend supported: tensorflow.compat.v1, tensorflow"""
```
Comment (lululxvi) [inline examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py:1]: Did you check both backends?

File: examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import deepxde as dde
+from matplotlib import pyplot as plt
+import numpy as np
+import skopt
+from skopt import gp_minimize
+from skopt.plots import plot_convergence, plot_objective
+from skopt.space import Real, Categorical, Integer
+from skopt.utils import use_named_args
+
+if dde.backend.backend_name == "pytorch":
```
Comment (lululxvi) [inline examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py:12]: Here, you have pytorch. Does the code support pytorch?

File: examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+search_result = gp_minimize(
+    func=fitness,
+    dimensions=dimensions,
+    acq_func="EI",  # Expected Improvement.
+    n_calls=n_calls,
+    x0=default_parameters,
+    random_state=1234,
+)
+
+search_result.x
```
Comment (lululxvi) [inline examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py:149]: print(search_result.x)

File: examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py (hunk 1 above)
Comment (pescap) [reply examples/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py]: yes

File: docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst [hunk 4]
```diff
@@ -0,0 +1,125 @@
+Helmholtz equation over a 2D square domain: Hyper-parameter tuning
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst:1]: Helmholtz equation over a 2D square domain: Hyperparameter optimization

File: docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+Notice that this script can be easilly adapted to other examples (either forward or inverse problems). 
+
+More scripts are available at `HPOMax <https://github.com/pescap/HPOMax>`_ GitHub repository. 
+
+Problem setup
+--------------
+
+We consider the same setting as in `Helmholtz equation over a 2D square domain <https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/helmholtz.2d.dirichlet.html>`_. 
+
+We apply GP-based Bayesian optimization via ``scikit-optimize`` (see `documentation <https://scikit-optimize.github.io/stable/>`_) over 50 calls. We use the *`Expected Improvement <https://scikit-optimize.github.io/stable/modules/generated/skopt.acquisition.gaussian_ei.html?highlight=ei#skopt.acquisition.gaussian_ei>_`* as acquisition function, define the minimum test error for each call as the (outer) loss function for the HPO.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst:17]: The link format of Expected Improvement is not correct, see https://deepxde--1109.org.readthedocs.build/en/1109/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.html

File: docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+We optimize the following hyper-parameters:
+
+- Learning rate: :math:`\alpha`;
```
Comment (lululxvi) [inline docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst:21]: Learning rate: --> Learning rate

File: docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst (hunk 4 above)
Comment (pescap) [reply docs/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.rst:1]: I edited it with "Hyper-parameter"
